artifactory/commands/terraform/terraformpublish.go

```diff
@@ -0,0 +1,304 @@
+	tpc.namespace = namespace
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) SetModuleName(moduleName string) *TerraformPublishCommand {
+	tpc.moduleName = moduleName
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) SetProvider(provider string) *TerraformPublishCommand {
+	tpc.provider = provider
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) SetTag(tag string) *TerraformPublishCommand {
+	tpc.tag = tag
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) SetExclusions(exclusions []string) *TerraformPublishCommand {
+	tpc.exclusions = exclusions
+	return tpc
+}
+
+func (tpc *TerraformPublishCommand) Run() error {
+	log.Info("Running Terraform Publish")
+	err := tpc.preparePrerequisites()
+	if err != nil {
+		return err
+	}
+	err = tpc.publish()
+	if err != nil {
+		return err
+	}
+	log.Info("Terraform publish finished successfully.")
+	return nil
+}
+
+func (tpc *TerraformPublishCommand) preparePrerequisites() error {
+	namespace, provider, tag, exclusions, err := ExtractTerraformPublishOptionsFromArgs(tpc.args)
+	if err != nil {
+		return err
+	}
+	if namespace == "" || provider == "" || tag == "" {
+		return errorutils.CheckErrorf("Wrong number of arguments. for a terraform publish command please provide --namespace, --provider and --tag.")
+	}
+	tpc.SetNamespace(namespace).SetProvider(provider).SetTag(tag).SetExclusions(exclusions)
+	if err := tpc.setRepoFromConfiguration(); err != nil {
+		return err
+	}
+	artDetails, err := tpc.serverDetails.CreateArtAuthConfig()
+	if err != nil {
+		return err
+	}
+	return utils.ValidateRepoExists(tpc.repo, artDetails)
+}
+
+func (tpc *TerraformPublishCommand) publish() error {
+	log.Debug("Deploying terraform module.")
+	success, failed, err := tpc.TerraformPublish()
+	if err != nil {
+		return err
+	}
+	tpc.result.SetSuccessCount(success)
+	tpc.result.SetFailCount(failed)
+	return nil
+}
+
+func ExtractTerraformPublishOptionsFromArgs(args []string) (namespace, provider, tag string, exclusions []string, err error) {
+	// Extract namespace information from the args.
+	flagIndex, valueIndex, namespace, err := coreutils.FindFlag("--namespace", args)
+	if err != nil {
+		return
+	}
+	coreutils.RemoveFlagFromCommand(&args, flagIndex, valueIndex)
+	// Extract provider information from the args.
+	flagIndex, valueIndex, provider, err = coreutils.FindFlag("--provider", args)
+	if err != nil {
+		return
+	}
+	coreutils.RemoveFlagFromCommand(&args, flagIndex, valueIndex)
+	// Extract tag information from the args.
+	flagIndex, valueIndex, tag, err = coreutils.FindFlag("--tag", args)
+	if err != nil {
+		return
+	}
+	coreutils.RemoveFlagFromCommand(&args, flagIndex, valueIndex)
+	// Extract exclusions information from the args.
+	flagIndex, valueIndex, exclusionsString, err := coreutils.FindFlag("--exclusions", args)
+	for _, singleValue := range strings.Split(exclusionsString, ";") {
+		exclusions = append(exclusions, singleValue)
+	}
+	if err != nil {
```

Comment:
This error check should be moved and be eight after the function which returned the error.